pkg/asset/machines/openstack/machines.go

```diff
@@ -36,20 +37,27 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 		return nil, fmt.Errorf("non-OpenStack machine-pool: %q", poolPlatform)
 	}
 	platform := config.Platform.OpenStack
-	mpool := pool.Platform.OpenStack
+
+	az := ""
+	trunk := platform.TrunkSupport
+
+	provider := generateProvider(clusterID, platform, pool.Platform.OpenStack, osImage, az, role, userDataSecret, trunk)
+
+	if role == "master" {
+		sg, err := createServerGroup(platform.Cloud, clusterID+"-"+role, "soft-anti-affinity")
```

Comment:
We should probably take the creation of server groups out of the `Machines` function, since it's called twice:
https://github.com/openshift/installer/blob/50ec977031fa86b60c78dfc3873f8e1d037d582c/pkg/asset/cluster/tfvars.go#L355
https://github.com/openshift/installer/blob/50ec977031fa86b60c78dfc3873f8e1d037d582c/pkg/asset/machines/master.go#L233

Reply:
AFAICS, the function called in `installer/pkg/asset/cluster/tfvars.go`:355 is just parsing the YAML spec:

https://github.com/openshift/installer/blob/50ec977031fa86b60c78dfc3873f8e1d037d582c/pkg/asset/machines/master.go#L439-L479

I could not find where the actual `openstack.Machines(string, *types.InstallConfig, *types.MachinePool, string, string, string)` function is called twice...

Can it be that Master.Generate is called twice?